external/llvm-project/mlir/lib/Conversion/ArithToAMDGPU/ArithToAMDGPU.cpp

```diff
@@ -204,7 +205,8 @@ LogicalResult TruncFToFloat8RewritePattern::match(arith::TruncFOp op) const {
   if (inType && inType.getWidth() <= 8 && saturateFP8)
     // Conversion between 8-bit floats is not supported with truncation enabled.
     return failure();
-  return success(outType.isFloat8E5M2FNUZ() || outType.isFloat8E4M3FNUZ());
+  return success(outType.isFloat8E5M2FNUZ() || outType.isFloat8E4M3FNUZ() ||
```

Comment:
I'd add chipset version guards here too

Reply:
That turned out to be a bit more elaborate, since that pass didn't already keep chipset around, but the changes were mostly easily copied from the other pass.